rust/otap-dataflow/crates/otap/src/otap_exporter.rs

```diff
@@ -218,22 +218,31 @@ impl local::Exporter<OtapPdata> for OTAPExporter {
                     }
                     //send data
                     Message::PData(pdata) => {
-                        // Capture signal type before moving pdata into try_from
-                        let signal_type = pdata.signal_type();
-
-                        self.pdata_metrics.inc_consumed(signal_type);
+                        // TODO(#1098): Note context is dropped.
                         let (_context, payload) = pdata.into_parts();
 
-                        // TODO(#1098): Note context is dropped.
-                        let message: OtapArrowRecords = payload
-                            .try_into()
-                            .inspect_err(|_| self.pdata_metrics.inc_failed(signal_type))?;
```

Comment:
I think we need to keep this call to try_into right? because if the payload was in otlp format, we want to convert it to OTAP when we call try_into here

Reply:
Good point! I could potentially add dedicated branches for all combinations of signal types and OtlpProtoBytes but the code doesn't look as good if do that. Not sure if all that is worth avoiding the cost of an additional match statement,  I'm going to close this PR for now. We could revisit this later.